scripttest/script_test_runner.go

```diff
@@ -523,89 +523,61 @@ func (st *scriptTest) verifyRemainingData(require *require.Assertions) {
-	//}, 10*time.Second, 50*time.Millisecond)
-	//require.NoError(err)
-	//if !ok2 {
-	//	for _, s := range st.testSuite.tektiteCluster {
-	//		keyStart := encoding.AppendUint64ToBufferBE(nil, common.UserSlabIDBase)
-	//		iter, err := s.GetStore().NewIterator(keyStart, nil, math.MaxUint64, false)
-	//		require.NoError(err)
-	//
-	//		for {
-	//			valid, err := iter.IsValid()
-	//			require.NoError(err)
-	//			if !valid {
-	//				break
-	//			}
-	//			k := iter.Current().Key
-	//			slabID := binary.BigEndian.Uint64(k[16:])
-	//			if slabID >= common.UserSlabIDBase {
-	//				ver := math.MaxUint64 - binary.BigEndian.Uint64(k[len(k)-8:])
-	//				if doLog {
-	//					log.Errorf("node: %d remaining entry for slab id %d %v:%v version:%d", nodeID, slabID, k, iter.Current().Value, ver)
-	//				}
-	//				hasUserEntries = true
-	//			}
-	//			err = iter.Next()
-	//			require.NoError(err)
-	//		}
-	//	}
-	//}
-	//require.True(ok2)
-}
-
-func checkRemainingData(require *require.Assertions, nodeID int, store *store.Store, keyStart []byte) bool {
-	iter, err := store.NewIterator(keyStart, nil, math.MaxUint64, true)
-	require.NoError(err)
-	// Should be no user data
-	hasUserEntries := false
-	for {
-		valid, err := iter.IsValid()
+	// We check there's no remaining user data in the store. Note that tombstones are written on the owning processor
+	// and may not have been pushed to the LSM yet, so we might expect to see user data if we create the iterator on
+	// a node different to the one that owns the processor. This is ok the tombstone will eventally be flushed to the LSM.
+	// So, if we detect data for a user partition we check on the owning node that there is no data.
+	for _, s := range st.testSuite.tektiteCluster {
+		store := s.GetStore()
+		iter, err := store.NewIterator(nil, nil, math.MaxUint64, false)
 		require.NoError(err)
-		if !valid {
-			break
-		}
-		k := iter.Current().Key
-		v := iter.Current().Value
-		slabID := binary.BigEndian.Uint64(k[16:])
-		if slabID >= common.UserSlabIDBase {
-			ver := math.MaxUint64 - binary.BigEndian.Uint64(k[len(k)-8:])
-			isPrefixTombstone := len(k) == 32 && len(v) == 0
-			isEndMarker := (len(k) == 33) && (len(v) == 1) && (v[0] == 'x')
-			if !isPrefixTombstone && !isEndMarker {
-				hasUserEntries = true
-				log.Errorf("%s node: %d remaining entry for slab id %d %v:%v version:%d", nodeID, slabID, k, v, ver)
-			}
-			if isPrefixTombstone {
-				log.Infof("node: %d got tombstone prefix for slab id %d : %v", nodeID, slabID, k)
+		for {
+			valid, err := iter.IsValid()
+			require.NoError(err)
+			if !valid {
+				break
 			}
-			if isEndMarker {
-				log.Infof("node: %d got tombstone end marker for slab id %d : %v", nodeID, slabID, k)
+			k := iter.Current().Key
+			slabID := binary.BigEndian.Uint64(k[16:])
+			if slabID >= common.UserSlabIDBase {
+				log.Infof("found remaining data for slab %d", slabID)
+				// There is user data
+				// Find which processor and node owns this prefix
+				partitionHash := k[:16]
```

Comment:
Why did you call it here partitionHash and on line 541 slabID?

Reply:
Note, the position of the colon is different.
Partition hash is the first 16 bytes.
Slab ID starts at position 16.